modules/storages/spec/models/file_link_spec.rb

```diff
@@ -0,0 +1,86 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2021 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require_relative '../spec_helper'
+
+describe ::Storages::FileLink, type: :model do
+  let(:creator) { create(:user) }
+  let(:project) { create(:project) } # , enabled_module_names: %i[storages work_packages]
+  let(:work_package) { create(:work_package, project: project) }
+  let(:storage) { create(:storage) }
+  let(:attributes) do
+    {
+      storage: storage,
+      creator: creator,
+      container_id: work_package.id,
+      container_type: "WorkPackage",
+      origin_id: "123456",
+      origin_name: "Origin Name",
+      origin_created_by_name: "Peter Pan",
+      origin_last_modified_by_name: "Lucy Lectric",
+      origin_mime_type: "text/html" }
+  end
+
+  describe '#create' do
+    it "creates an instance" do
+      file_link = described_class.create attributes
+      expect(file_link).to be_valid
+    end
+
+    it "create instance should fail with unsupported container type" do
```

Comment:
keep English grammar. Humans have to read this.
```suggestion
    it "should fail when creating an instance with unsupported container type" do
```